Source/automap.cpp

```diff
@@ -188,9 +189,9 @@ void __cdecl AutomapZoomIn()
 		AutoMapScale += 5;
 		AutoMapPosBits = (AutoMapScale << 6) / 100;
 		AutoMapXPos = AutoMapPosBits >> 1;
-		AutoMapYPos = AutoMapPosBits >> 2;
-		AMPlayerX = AutoMapPosBits >> 3;
-		AMPlayerY = AutoMapPosBits >> 4;
+		AutoMapYPos = AutoMapXPos >> 1;

```

Comment:
This code seems strange. Is it really correct?

Should it not be as follows:

```c
 		AutoMapYPos = AutoMapYPos >> 1; 
 		AMPlayerX = AMPlayerX >> 1; 
 		AMPlayerY = AMPlayerY >> 1; 
```

Reply:
It's correct. Each of those variables is assigned half the value of the one before it. I'm pretty sure the names are incorrect (they make no sense) and I was planning to change AutomapPosBits into an array instead.

Reply:
Those variable names are likely incorrect. While they line up with the PSX version, the PSX renders the automap entirely different as it does not support zooming. Each one of those is a different zoom factor, and not player/map coordinates.

Reply:
Also they get moved around by the linker, an array assures they stay in order.

Reply:
Where did those automap_zoom_32/16/8/4/2 names come from?

Reply:
> Where did those automap_zoom_32/16/8/4/2 names come from?

We made them up, as it seemed to fit the context. basically zoom level divided by *n*.

```c
    automap_zoom_32 = (automap_zoom_level << 6) / 100;
    automap_zoom_16 = automap_zoom_32 >> 1;
    automap_zoom_8 = automap_zoom_32 >> 2;
    automap_zoom_4 = automap_zoom_32 >> 3;
    automap_zoom_2 = automap_zoom_32 >> 4;
```

Which is equivalent to:

```c
    automap_zoom_32 =(automap_zoom_level * 64) / 100;
    automap_zoom_16 = automap_zoom_32 / 2;
    automap_zoom_8 = automap_zoom_32 / 4;
    automap_zoom_4 = automap_zoom_32 / 8;
    automap_zoom_2 = automap_zoom_32 / 16;
```

So perhaps a better name would be `automap_zoom_64` instead of `automap_zoom_32`, since as it is 64% of the zoom_level? That would give us.

```c
    automap_zoom_64 =(automap_zoom_level * 64) / 100;
    automap_zoom_32 = automap_zoom_64 / 2;
    automap_zoom_16 = automap_zoom_64 / 4;
    automap_zoom_8 = automap_zoom_64 / 8;
    automap_zoom_4 = automap_zoom_64 / 16;
```